### ISSUE TITLE
workarounds for lighttpd

### ISSUE THREAD

**Issue by jmtd** [httpoxy/httpoxy-org#5]
as usual lighttpd is lagging and there doesn't appear to be a way to fix this properly, but this workaround *might* be sufficient

```
setenv.add-request-header = (
"Proxy" => "invalid",
)
```

results in e.g.

```
$ GET -H "Proxy: Foo" <some-server>
HTTP_PROXY=Foo, invalid
```

**Comment by dominics** [comment]
That'd possibly cause problems for the outgoing requests if they didn't parse the HTTP_PROXY and correctly reject invalid URLs (all outgoing requests would start to fail?)

Probably still better than actually being vulnerable though.

**Comment by drdaxxy** [comment]
Lighttpd users whose binary has been built with Lua support can install mod_magnet and put something this in their configuration to block requests with the Proxy header set:

```
server.modules += ( "mod_magnet" )
magnet.attract-raw-url-to = ( "/path/to/mitigate-httpoxy.lua" )
```
`mitigate-httpoxy.lua` containing:
```
if lighty.request["Proxy"] then
  return 403
end
```

Before anyone asks: yes, the lookup of request["Proxy"] is case-insensitive, I checked.

This *should* catch every request as far as I'm aware, but I'm not an expert on lighty internals so I don't know where in the pipeline mod_magnet sits (and if it's possible for requests to leave lighty before they get to mod_magnet). I've tested this with a FastCGI handler, but not a setup where lighty is used as a reverse proxy or anything.

Relevant documentation: https://redmine.lighttpd.net/projects/lighttpd/wiki/Docs_ModMagnet
